### PR TITLE
updated text processing

### PR DESCRIPTION
**Pull request by Laralinmcc**

### CODE REVIEW

File: cur/programming/python/hug_text_processing/python_building_a_markov_model.html [hunk 1]
```diff
@@ -60,7 +60,23 @@
...
+    """ Returns the markov model for text.
+    m is the model (a dictionary)
+    text is the text
+    k is the order of our Markov chain"""</code>
+    </pre>
+
+<p> This function should loop through text, process each character and add it to the model accordingly. For example calling build_markov_model('the ether', 3) would return {' et': {'h': 1}, 'the': {' ': 1, 'r': 1}, 'e e': {'t': 1}, 'he ': {'e': 1}, 'eth': {'e': 1}}.</p>
+
+<p class="alert quoteBlue">
+Hint: Use the process_character function in your implementation of build_markov_model</code>.
```
Comment (cycomachead) [inline cur/programming/python/hug_text_processing/python_building_a_markov_model.html:78]: Function names should be inside `<code></code>` blocks.

File: cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html [hunk 2]
```diff
@@ -5,8 +5,92 @@
...
+
+<pre>
+<code>text = "the theremin in the basement is the theologian's theft detection device."
+m = construct_markov_model(text, 3)
+random_character(m, "the")      would return " " with probability 3/6, and "r", "o", or "f" with probability 1/6
+random_character(m, "ase")      would return "m" with probability 1</code>
+</pre>
+
+
+        <p>In Snap! we have a block that can pick random numbers and a blocks that can pick a random item from a list. Python also has these functions, but in order to access these functions we must import the library that contains them. This is just like importing a library in Snap! At the top of your markov.py file add the following line:</p>
```
Comment (cycomachead) [inline cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html:48]: `Snap<em>!</em>`

File: cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html [hunk 3]
```diff
@@ -5,8 +5,92 @@
...
+
+        <pre>
+            <code>import random</code>
+        </pre>
+
+        <p class="alert quoteBlue">
+        Hint: Look at the python random library's <a href="https://docs.python.org/2/library/random.html">documentation</a> for helpful functions.
```
Comment (cycomachead) [inline cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html:55]: Shouldn't this be Python 3 docs? (though the differences are probably minor)

File: cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html [hunk 4]
```diff
@@ -5,8 +5,92 @@
...
+
+    m is a Markov Model
+    N is the desired output length"""</code>
+    </pre>
+
+        <p><code>generate_random_text</code> should start by picking a random kgram from among the possible kgrams that appear in the text. This will form the first k characters of the randomly generated output text. This first choice does not have to obey any particular statistical properties. We observe from the example above that the kgram selected was "uest,\nth". We call this the seed-kgram.</p>
+
+        <p>After the first kgram is selected, the program should use random_character to select the next character. In the example above, the first such character selected was "o". The program should then use this character to form a new seed-kgram of length k by throwing away the 0th character of the old seed, and appending the new character to the end. In the example above, the seed k-gram becomes "est,\ntho", and the next character to be selected will be selected using the distribution for this seed-kgram. In this case the next character selected is "u", and the new seed-kgram becomes "st,\nthou". This process is repeated until we have 80 total characters.</p>
```
Comment (cycomachead) [inline cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html:71]: `<code></code>`missing

File: cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html [hunk 5]
```diff
@@ -5,8 +5,92 @@
...
+</pre>
+        <p> The output after running the above lines of code would look something like this:</p>
+<pre>
+<code>uest,
+thought of an inch; slip your hoe-cake done!"
+"Seven hundred feet above all ot</code>
+</pre>
+
+        <p class="alert quoteBlue">
+        Note: The output is generated random so your output will not be the same as the shown output above.
```
Comment (cycomachead) [inline cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html:88]: "random" → "randomly"

File: cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html [hunk 6]
```diff
@@ -5,8 +5,92 @@
...
         </p>
 
+        <p>Because the new kgram obeys the same statistical properties as the original text, you know that any kgram that is generated is guaranteed to appear at some point in the text.</p>
+
+        <p>Try generating texts using various ks. You should observe that for small k (k= 1, 2, 3), the generated text resembles but is not quite English (or whatever language utilized by your modeled text). For higher order k, the text should start looking more and more like the original text.</p>
```
Comment (cycomachead) [inline cur/programming/python/hug_text_processing/python_generating_text_with_a_markov_model.html:93]: `k` should probably be in  a `<code></code>`.

File: cur/programming/python/hug_text_processing/python_building_a_markov_model.html (hunk 1 above)
Comment (Laralinmcc) [inline cur/programming/python/hug_text_processing/python_building_a_markov_model.html]: Separate from the comment?

**Pull request merged**